CMAKE VERSION ERROR

The `CMakeLists.txt` checks for a CMake version that is no longer compatible with current CMake release.

```
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.

```
It can be fixed by changing:
```                                                                              
cmake_minimum_required (VERSION 3.1.0 FATAL_ERROR)
cmake_policy(VERSION 3.1.0)
```
to
```                                                                    
cmake_minimum_required (VERSION 3.6.0 FATAL_ERROR)
cmake_policy(VERSION 3.6.0)
```
in `CMakeLists.txt`

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,12 @@
-cmake_minimum_required (VERSION 3.1.0 FATAL_ERROR)
-cmake_policy(VERSION 3.1.0)
+if (${CMAKE_VERSION} VERSION_LESS 4.0)
+  cmake_minimum_required (VERSION 3.1.0 FATAL_ERROR)
+  cmake_policy(VERSION 3.1.0)
+else()
+  # CMake 4.0 does not allow compatibility with CMake < 3.5
+  cmake_minimum_required (VERSION 4.0 FATAL_ERROR)
+  cmake_policy(VERSION 4.0)
+endif()
+
 
 project (photospline VERSION 2.4.0 LANGUAGES C CXX)
 

